factcast-snapshotcache-mongodb/src/main/java/org/factcast/core/snap/mongo/MongoDbSnapshotCache.java

```
@@ -0,0 +1,151 @@
+        aggregateIdIndexResult);
+
+    // Third index for TTL management of documents, expires the document after 0 seconds of the
+    // `expireAt` field. So, immediately.
+    collection.createIndex(
+        Indexes.ascending(EXPIRE_AT_FIELD), new IndexOptions().expireAfter(0L, TimeUnit.SECONDS));
+  }
+
+  @Override
+  public @NonNull Optional<SnapshotData> find(@NonNull SnapshotIdentifier id) {
+    Document query = getDocumentById(id);
+
+    Document result = collection.find(query).first();
+
+    if (result != null) {
+      Binary binary = result.get(SERIALIZED_PROJECTION_FIELD, Binary.class);
+      byte[] bytes = binary.getData();
+      String serializerId = result.getString(SNAPSHOT_SERIALIZER_ID_FIELD);
+      UUID lastFactId = UUID.fromString(result.getString(LAST_FACT_ID_FIELD));
+
+      tryUpdateExpirationDate(id);
+
+      return Optional.of(
+          new SnapshotData(bytes, SnapshotSerializerId.of(serializerId), lastFactId));
+    }
+
+    return Optional.empty();
+  }
+
+  private void tryUpdateExpirationDate(SnapshotIdentifier id) {
+    try {
+      collection.updateOne(
+          getDocumentById(id),
+          Updates.set(
+              EXPIRE_AT_FIELD,
+              Instant.now().plus(properties.getDeleteSnapshotStaleForDays(), ChronoUnit.DAYS)));
+    } catch (Exception e) {
+      log.warn("Failed to update expiration date for snapshot with id: {}", id, e);
+    }
+  }
+
+  @Override
+  public void store(@NonNull SnapshotIdentifier id, @NonNull SnapshotData snapshot) {
+    Binary bytes = new Binary(BsonBinarySubType.BINARY, snapshot.serializedProjection());
+
+    Document doc =
+        new Document(PROJECTION_CLASS_FIELD, id.projectionClass().getName())
+            .append(SNAPSHOT_SERIALIZER_ID_FIELD, snapshot.snapshotSerializerId().name())
+            .append(LAST_FACT_ID_FIELD, snapshot.lastFactId().toString())
+            // Document limit for binary data(16MB)
```

what if?
i mean we don't even test it, but just let it fail?
As this is not only used by aggregates, but also full blown snapshotprojections, we cannot predict the amount of data contained.

GridFS seems supported (if the blob exceeds its limits): https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html